src/pages/2018-11-21-bull/index.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Processing jobs across microservices using Bull 🐂
+date: '2018-11-21T22:12:03.284Z'
+path: '/job-processor-async/'
+---
+
+> An in-depth explanation into why we invested in the open source package, <a href="https://github.com/OptimalBits/bull" target="_blank">Bull</a>, even though it didn't quite fit our needs.
+
+Into the microservice world we go, exploring where no developer has gone before. When switching to a service oriented stack versus the ol' monolith, new problems arise. The simple async job processor of the past doesn't fit in this new architecture. We found <a href="https://github.com/OptimalBits/bull" target="_blank">Bull</a>, a Node.js package, to cover our needs, but needed tweaks to work in this new world.
+
+If you have one service which performs a task, e.g. an image processor, and another service which needs to queue jobs to it, we cannot use traditional technology. I am of course mentioning things like DelayedJob or Celery which you may be familiar with. Keeping it 2018, this Node.js package is lightning fast, built to work with Redis and has an active community. It didn't quite fit our needs at first as it processed jobs in the same app as the queue-mechanism. This is fine for traditional apps, but for our setup we needed to manage jobs **across systems**.
```

Comment:
Hmm I don't quite get why "traditional" methods like DelayedJob / Celery cannot be used. Would you be able to illustrate w/ a concrete example?